src/components/widgets/AssocListPopup/AssocListPopup.tsx

```diff
@@ -56,16 +70,79 @@ export const AssocListPopup: FunctionComponent<IAssocListProps & IAssocListActio
         onClose()
     }, [onSave, onClose])
 
+    const filterData = React.useCallback(() => {
+        const pendingDataValue = props.pendingDataChanges[props.widget.bcName]
+        const filterValue: string[] = []
+        for (const value in pendingDataValue) {
+            if (pendingDataValue[value]?.id && pendingDataValue[value]?._associate === true) {
+                filterValue.push(pendingDataValue[value].id.toString())
+            }
+        }
+        onFilter(props.calleeBCName, {
+            type: FilterType.equalsOneOf,
+            fieldName: props.associateFieldKey,
+            value: filterValue
+        })
+        onClose()
+    }, [onFilter, onClose, props.pendingDataChanges])
+
     const cancelData = React.useCallback(() => {
         onCancel()
         onClose()
     }, [onCancel, onClose])
 
+    const handleDeleteTag = React.useCallback((val: DataItem) => {
+        onDeleteTag(props.widget.bcName, val)
+    },
+        [props.onDeleteTag]
+    )
+
+    const pendingCurrentData = props.pendingDataChanges[props.widget.bcName]
+    const pendingData = []
+    for (const key in pendingCurrentData) {
+        if (pendingCurrentData[key]._associate === true) {
+            pendingCurrentData[key].closable = true
+            pendingData.push(pendingCurrentData[key])
+        }
+    }
+
+    // Tag values limit
+    const tagLimit = 5
+    const tagBackgroundCount: number = pendingData.length - tagLimit
+    if (pendingData.length > tagLimit) {
+        pendingData.length = tagLimit
```

Comment:
Please don't use this pattern to trim the array. It will not lead to a problems here, but if something change with how we fill `pendingData` in the code above (pass it from parent or store for example), we have a very high chances to break something as setting `length` works by reference and may modify some external array.
Prefer `.slice`:
```tsx
const visiblePendingData = pendingData.length > tagLimit
    ? [ ...pendingData.slice(0, tagLimit), { id: '9' } ]
      : pendingData
```

Reply:
fixed